pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -336,6 +347,56 @@ func (p *PriorityQueue) Run(logger klog.Logger) {
 	}, 30*time.Second, p.stop)
 }
 
+// isPodWorthRequeuing calls QueueingHintFn of only plugins registered in pInfo.unschedulablePlugins.
+// If any QueueingHintFn returns QueueImmediately, the scheduling queue is supposed to enqueue this Pod to activeQ.
+// If no QueueingHintFn returns QueueImmediately, but some return QueueAfterBackoff,
+// the scheduling queue is supposed to enqueue this Pod to activeQ/backoffQ depending on the remaining backoff time of the Pod.
+// If all QueueingHintFn returns QueueSkip, the scheduling queue enqueues the Pod back to unschedulable Pod pool
+// because no plugin changes the scheduling result via the event.
+func (p *PriorityQueue) isPodWorthRequeuing(logger klog.Logger, pInfo *framework.QueuedPodInfo, event framework.ClusterEvent, oldObj, newObj interface{}) framework.QueueingHint {
+	if pInfo.UnschedulablePlugins.Len() == 0 || event.IsWildCard() {
+		logger.V(6).Info("Worth requeuing because no unschedulable plugins", "pod", klog.KObj(pInfo.Pod))
```

Comment:
was this existing logic?

Also, maybe worth different messages for empty `UnschedulablePlugins` and the wildcard.

Reply:
Yes. In both cases, we always enqueue Pod to activeQ/backoffQ now.

- When `pInfo.UnschedulablePlugins.Len() == 0`
  - https://github.com/kubernetes/kubernetes/blob/336e46101c4451f504c884d65bb7b7b5f6f7f4d8/pkg/scheduler/internal/queue/scheduling_queue.go#L778
- When `event.IsWildCard()`
  - `podMatchesEvent` always returns `true` https://github.com/kubernetes/kubernetes/blob/336e46101c4451f504c884d65bb7b7b5f6f7f4d8/pkg/scheduler/internal/queue/scheduling_queue.go#L1151-L1153
  - and then https://github.com/kubernetes/kubernetes/blob/336e46101c4451f504c884d65bb7b7b5f6f7f4d8/pkg/scheduler/internal/queue/scheduling_queue.go#L778


> Also, maybe worth different messages for empty UnschedulablePlugins and the wildcard.

👍